src/utils.ts

```diff
@@ -104,8 +104,17 @@ class Utils {
 		const platformConfig = this.getPlatformConfig(platformId);
 		const options: QueueSettings = platformConfig.queue.defaults;
 		if (!options.redis) { options.redis = {}; }
-		options.redis.host = process.env.REDIS_HOST || '127.0.0.1';
-		options.redis.port = parseInt(process.env.REDIS_PORT || '6379', undefined);
+		if (!options.redis.url) {
+			options.redis.host = process.env.REDIS_HOST || '127.0.0.1';
+			options.redis.port = parseInt(process.env.REDIS_PORT || '6379', undefined);
+		}
+		options.redis.retry_strategy = (opts) => {
+			if (opts.attempt === 3) {
+				logger.error('Unable to connect to the Redis server - Notification Service is going to exit!');
+				process.exit(1);
+			}
+			return 1000;
```

Comment:
I think next retriy shoud be (opts.attempt+1)*1000. That should be performed an newly attempted connections spread over a period of time.